gateway/src/main/java/com/hurence/webapiservice/historian/impl/SolrHistorianServiceImpl.java

```diff
@@ -341,6 +347,50 @@ public HistorianService getAnnotations(JsonObject params, Handler<AsyncResult<Js
         return this;
     }
 
+    @Override
+    public HistorianService addTimeSeries(JsonArray timeseries, Handler<AsyncResult<JsonObject>> resultHandler) {
+
+        Handler<Promise<JsonObject>> getMetricsNameHandler = p -> {
+            try {
+                JsonObject response = new JsonObject();
+                Collection<SolrInputDocument> documents = new ArrayList<>();
+                int numChunk = 0;
+                int numPoints = 0;
+                for (Object TimeseriesObject : timeseries) {
+                    JsonObject timeserie = (JsonObject) TimeseriesObject;
+                    LOGGER.info("building SolrDocument from a chunk");
+                    SolrInputDocument document = chunkTimeSerie(timeserie);
+                    documents.add(document);
+                    int totalNumPointsInChunk = (int) document.getFieldValue(RESPONSE_CHUNK_SIZE_FIELD);
+                    numChunk++;
+                    numPoints = numPoints + totalNumPointsInChunk;
+                }
+                LOGGER.info("adding some chunks in collection {}", solrHistorianConf.chunkCollection);
+                solrHistorianConf.client.add(solrHistorianConf.chunkCollection, documents);
+                solrHistorianConf.client.commit(solrHistorianConf.chunkCollection);
+                LOGGER.info("added with success some chunks in collection {}", solrHistorianConf.chunkCollection);
+                String message = "injected " + numPoints + " points of " + numChunk + " metrics in " + numChunk + " chunks";
```

Comment:
DO you handle the case when there are 0 points injected ? Same before building a chunk ensure it is not empty. What happens if client send a request with metric names but empty array ? I think this should be tested in the http handler, When you are parsing the request ensure there is points for each metric.

Reply:
done

Reply:
When I said handling the case there is 0 point I just meant to ensure there is no NPE. Not  to throw an error in this case.